### PR TITLE
Fixed yes, no changing to boolean

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/23825
https://github.com/demisto/etc/issues/23948
https://github.com/demisto/etc/issues/23448
https://github.com/demisto/etc/issues/23937

## Screenshots
image
image

## Must have
- [x] Tests

### CODE REVIEW

File: demisto_sdk/commands/format/update_generic_yml.py [hunk 1]
```diff
@@ -70,11 +70,12 @@ def repr_str(dumper, data):
             return dumper.org_represent_str(data)
 
         yaml.add_representer(str, repr_str, Dumper=yamlordereddictloader.SafeDumper)
-
         with open(self.output_file, 'w') as f:
-            ryaml.dump(
+            round_trip_dump(
```
Comment (Itay4) [inline demisto_sdk/commands/format/update_generic_yml.py:74]: lets make it a one-liner

File: demisto_sdk/commands/format/tests/test_formatting_yml_test.py [hunk 2]
```diff
@@ -161,3 +161,37 @@ def test_pwsh_format(tmpdir, yml_file, yml_type):
...
+    (SOURCE_FORMAT_PLAYBOOK_COPY, DESTINATION_FORMAT_PLAYBOOK_COPY, PlaybookYMLFormat, 'File Enrichment-GenericV2_copy',
+     'playbook')
+]
+
+
+@pytest.mark.parametrize('source_path, destination_path, formatter, yml_title, file_type', PLAYBOOK_TEST)
+def test_string_condition_in_playbook(source_path, destination_path, formatter, yml_title, file_type):
+    """
+    Given
+    - Valid playbook yml source_path.
```
Comment (Itay4) [inline demisto_sdk/commands/format/tests/test_formatting_yml_test.py:176]:
```suggestion
    - Playbook with condition labeled as `yes`.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 * Fixed a bug in the **download** command when custom content contained not supported content entity.
+* Fixed a bug in **format** command when the command switched 'yes', 'no' string to boolean.
```
Comment (Itay4) [inline CHANGELOG.md:3]:
```suggestion
* Fixed a bug in **format** command in which boolean strings  (e.g. 'yes' or 'no') were converted to boolean values (e.g. 'True' or 'False').
```

File: demisto_sdk/commands/format/update_playbook.py [hunk 4]
```diff
@@ -76,6 +76,7 @@ def run_format(self) -> int:
             self.update_tests()
             self.update_conf_json('playbook')
             self.add_description()
+            self.delete_sourceplaybookid()
```
Comment (Itay4) [inline demisto_sdk/commands/format/update_playbook.py:79]: please add test for that

Comment (Shellyber) [reply demisto_sdk/commands/format/update_playbook.py:79]: There is already test for that.
check in demisto_sdk/commands/format/tests/test_formatting_yml_test.py

def test_playbook_sourceplaybookid(source_path):
    schema_path = os.path.normpath(
        os.path.join(__file__, "..", "..", "..", "common", "schemas", '{}.yml'.format('playbook')))
    base_yml = PlaybookYMLFormat(source_path, path=schema_path)
    base_yml.delete_sourceplaybookid()

    assert 'sourceplaybookid' not in base_yml.data

Comment (Itay4) [reply demisto_sdk/commands/format/update_playbook.py:79]: so how come it did not fail until now ?

**Comment by Shellyber** [comment]
@Itay4 Done. please review

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,9 @@
 # Changelog
 * Added formatting to **generate-docs** command results replacing all <br> tags with <br/>.
 * Fixed a bug in the **download** command when custom content contained not supported content entity.
+* Fixed a bug in **format** command in which boolean strings  (e.g. 'yes' or 'no') were converted to boolean values (e.g. 'True' or 'False').
+* **format** command now removes *sourceplaybookid* field.
```
Comment (Itay4) [inline CHANGELOG.md:5]:
```suggestion
* **format** command now removes *sourceplaybookid* field from playbook files.
```

File: demisto_sdk/tests/integration_tests/format_integration_test.py [hunk 6]
```diff
@@ -226,3 +227,36 @@ def _verify_conf_json_modified(test_playbooks: List, yml_title: str, conf_json_p
...
+
+
+PLAYBOOK_TEST = [
+    (SOURCE_FORMAT_PLAYBOOK_COPY, DESTINATION_FORMAT_PLAYBOOK_COPY, PlaybookYMLFormat, 'File Enrichment-GenericV2_copy',
+     'playbook')
+]
+
+
+@pytest.mark.parametrize('source_path, destination_path, formatter, yml_title, file_type', PLAYBOOK_TEST)
+def remove_playbook_sourceplaybookid(source_path, destination_path, formatter, file_type):
```
Comment (Itay4) [inline demisto_sdk/tests/integration_tests/format_integration_test.py:239]:
```suggestion
def test_integration_format_remove_playbook_sourceplaybookid(source_path, destination_path, formatter, file_type):
```

File: demisto_sdk/tests/integration_tests/format_integration_test.py [hunk 7]
```diff
@@ -226,3 +227,36 @@ def _verify_conf_json_modified(test_playbooks: List, yml_title: str, conf_json_p
...
         raise
+
+
+PLAYBOOK_TEST = [
+    (SOURCE_FORMAT_PLAYBOOK_COPY, DESTINATION_FORMAT_PLAYBOOK_COPY, PlaybookYMLFormat, 'File Enrichment-GenericV2_copy',
+     'playbook')
+]
+
+
+@pytest.mark.parametrize('source_path, destination_path, formatter, yml_title, file_type', PLAYBOOK_TEST)
```
Comment (Itay4) [inline demisto_sdk/tests/integration_tests/format_integration_test.py:238]: no need to parametrize here, lets just run it on one playbook

File: demisto_sdk/tests/integration_tests/format_integration_test.py [hunk 8]
```diff
@@ -226,3 +229,36 @@ def _verify_conf_json_modified(test_playbooks: List, yml_title: str, conf_json_p
...
+    - Playbook with field  `sourceplaybookid`.
+    - destination_path to write the formatted playbook to.
+
+    When
+    - Running the format command.
+
+    Then
+    - Ensure 'sourceplaybookid' was deleted from the yml file.
+    """
+    source_playbook_path = join(DEMISTO_SDK_PATH, "tests/test_files/format_new_playbook_copy.yml")
```
Comment (Itay4) [inline demisto_sdk/tests/integration_tests/format_integration_test.py:250]: use `SOURCE_FORMAT_PLAYBOOK_COPY` instead

File: demisto_sdk/tests/integration_tests/format_integration_test.py [hunk 9]
```diff
@@ -226,3 +229,36 @@ def _verify_conf_json_modified(test_playbooks: List, yml_title: str, conf_json_p
...
+    output_playbook_path = join(DEMISTO_SDK_PATH, "tests/test_files/new_format_new_playbook_copy.yml")
```
Comment (Itay4) [inline demisto_sdk/tests/integration_tests/format_integration_test.py:251]: please use `tmp_path` fixture instead 
it handles temp files better

File: demisto_sdk/tests/integration_tests/format_integration_test.py [hunk 10]
```diff
@@ -226,3 +230,36 @@ def _verify_conf_json_modified(test_playbooks: List, yml_title: str, conf_json_p
...
+
+    runner = CliRunner(mix_stderr=False)
+    arguments = [
+        'format',
+        '-i', source_playbook_path,
+        '-o', output_playbook_path
+    ]
+    runner.invoke(main, arguments)
+
```
Comment (Itay4) [inline demisto_sdk/tests/integration_tests/format_integration_test.py:260]: assert exit_code == 0
assert no exception
assert no stderr
assert that expected output (Starting updates... Finished updates..., etc..) is in stdout

**Pull request merged**